src/main/java/graphql/execution/preparsed/PreparsedDocumentProvider.java

```diff
@@ -7,20 +7,23 @@
 import java.util.function.Function;
 
 /**
- * Interface that allows clients to hook in Document caching and/or the whitelisting of queries
+ * Interface that allows clients to hook in Document caching and/or the whitelisting of queries.
  */
 @PublicSpi
 public interface PreparsedDocumentProvider {
     /**
-     * This is called to get a "cached" pre-parsed query and if its not present, then the computeFunction
-     * can be called to parse and validate the query
-     *
-     * @param executionInput  The {@link graphql.ExecutionInput} containing the query
-     * @param computeFunction If the query has not be pre-parsed, this function can be called to parse it
+     * This is called to get a "cached" pre-parsed query and if its not present, then the "parseAndValidateFunction"
+     * can be called to parse and validate the query.
+     * <p>
+     * Note - the "parseAndValidateFunction" MUST be called if you dont have a per parsed version of the query because it not only parses
+     * and validates the query, it invokes {@link graphql.execution.instrumentation.Instrumentation} calls as well for parsing and validation.
+     * if you dont make a call back on this then these wont happen.
      *
+     * @param executionInput           The {@link graphql.ExecutionInput} containing the query
+     * @param parseAndValidateFunction If the query has not be pre-parsed, this function MUST be called to parse and validate it
      * @return an instance of {@link PreparsedDocumentEntry}
      */
-    PreparsedDocumentEntry getDocument(ExecutionInput executionInput, Function<ExecutionInput, PreparsedDocumentEntry> computeFunction);
+    PreparsedDocumentEntry getDocument(ExecutionInput executionInput, Function<ExecutionInput, PreparsedDocumentEntry> parseAndValidateFunction);
```

Comment:
This is a much better name for the callback parameter because that's what it does